src/main/java/org/folio/entitlement/integration/kong/KongConfiguration.java

```diff
@@ -80,4 +95,30 @@ public KongModuleRouteUpdater kongModuleRouteUpdater(KongGatewayService kongGate
   public KongModuleRouteCleaner kongModuleRouteCleaner(KongGatewayService kongGatewayService) {
     return new KongModuleRouteCleaner(kongGatewayService);
   }
+
+
+  /**
+   * Creates a {@link org.springframework.cloud.openfeign.FeignClient} for integration with Kong Admin API.
+   *
+   * @param properties - kong configuration properties with required data
+   * @param contract - feign contract
+   * @param encoder - feign http body encoder
+   * @param decoder - feign http body decoder
+   * @return created {@link KongAdminClient} component
+   */
+  @Bean(name = "folioKongAdminClient")
+  public KongAdminClient folioKongIntegrationClient(okhttp3.OkHttpClient okHttpClient,
+    KongConfigurationProperties properties, Contract contract, Encoder encoder, Decoder decoder) {
+
+    var feignClientBuilder = Feign.builder().contract(contract).encoder(encoder).decoder(decoder).errorDecoder(
+      new FeignRetrySupportingErrorDecoder(new ErrorDecoder.Default(),
+        methodKeyAndResponse -> methodKeyAndResponse.getRight().status() == 500));
+
+    var numberOfRetries = properties.getRetries() != null ? properties.getRetries() : 3;
```

Comment:
this configuration is related to the Kong service and used in KongModuleRegistrar, not sure that reuse of it is a good approach:
```   
kongGatewayService.upsertService(
      new Service().name(moduleId).url(moduleUrl)
        .connectTimeout(properties.getConnectTimeout())
        .readTimeout(properties.getReadTimeout())
        .writeTimeout(properties.getWriteTimeout())
        .retries(properties.getRetries())
    );
```

Reply:
That retries config is never used.